frontend/src/app/main/data/workspace.cljs

```diff
@@ -1236,22 +1237,80 @@
                 (with-meta params
                   {:on-success image-uploaded})))))))
 
+(declare paste-text)
+
 (def paste
   (ptk/reify ::paste
     ptk/WatchEvent
     (watch [_ state stream]
-      (->> (wapi/read-from-clipboard)
-           (rx/map t/decode)
-           (rx/filter #(= :copied-shapes (:type %)))
-           (rx/map #(select-keys % [:selected :objects]))
-           (rx/map paste-impl)
-           (rx/catch (partial instance? js/SyntaxError)
-             (fn [_]
-               (->> (wapi/read-image-from-clipboard)
-                    (rx/map paste-image-impl))))
-           (rx/catch (fn [err]
-                       (js/console.error "Clipboard error:" err)
-                       (rx/empty)))))))
+      (try
+        (let [clipboard-str (wapi/read-from-clipboard)
+
+              paste-transit-str
+              (->> clipboard-str
+                   (rx/filter t/transit?)
+                   (rx/map t/decode)
+                   (rx/filter #(= :copied-shapes (:type %)))
+                   (rx/map #(select-keys % [:selected :objects]))
+                   (rx/map paste-impl))
+
+              paste-plain-text-str
+              (->> clipboard-str
+                   (rx/filter (comp not empty?))
+                   (rx/map paste-text))
+
+              paste-image-str
+              (->> (wapi/read-image-from-clipboard)
+                   (rx/map paste-image-impl))]
+
+          (->> (rx/concat paste-transit-str
+                          paste-plain-text-str
+                          paste-image-str)
+               (rx/first)
+               (rx/catch
+                   (fn [err]
+                     (js/console.error "Clipboard error:" err)
+                     (rx/empty)))))
+        (catch :default e
+          (.error js/console "ERROR" e))))))
+
+(defn as-content [text]
+  (let [paragraphs (->> (str/split text "\n")
```

Comment:
You can use `str/lines` here, that treats properly different line endings.

Reply:
OK